### PR TITLE
Bug 1379149: Optimize Contributors query

### PR DESCRIPTION
**Pull request by mathjazz**
@jotes r?

As mentioned in the [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1379149#c2), this is a simple optimization of the contributors query inspired by [bug 1357361](https://bugzilla.mozilla.org/show_bug.cgi?id=1357361).

I tested on stage and the entire /permissions page request gets around 60% shorter.

### CODE REVIEW

File: pontoon/teams/views.py [hunk 1]
```diff
@@ -122,13 +122,11 @@ def ajax_permissions(request, locale):
...
     all_users = User.objects.exclude(pk__in=managers).exclude(pk__in=translators).exclude(email='')
 
-    contributors = (
-        User.translators
-        .filter(translation__locale=l)
-        .values_list('email', flat=True)
-        .distinct()
-    )
+    contributors_qs = User.translators.with_translation_counts(None, Q(locale=l))
+    contributors = set([contributor.email for contributor in contributors_qs])
```
Comment (phlax) [inline pontoon/teams/views.py]: no need for brackets here you can you a generator expression rather than list comprehension

Comment (phlax) [reply pontoon/teams/views.py]: also might be more efficient to add `.iterator()` on the qs

Comment (mathjazz) [reply pontoon/teams/views.py]: Updated. `.iterator()` I cannot use because `with_translation_counts` returns a `list`.

**Comment by phlax** [comment]
lgtm

File: pontoon/teams/views.py [hunk 2]
```diff
@@ -122,13 +122,11 @@ def ajax_permissions(request, locale):
...
     all_users = User.objects.exclude(pk__in=managers).exclude(pk__in=translators).exclude(email='')
 
-    contributors = (
-        User.translators
-        .filter(translation__locale=l)
-        .values_list('email', flat=True)
-        .distinct()
-    )
+    contributors_qs = User.translators.with_translation_counts(None, Q(locale=l))
+    contributors = set(contributor.email for contributor in contributors_qs)
```
Comment (jotes) [inline pontoon/teams/views.py]: [nit] `{contributor.email for contributor...` could be a little shorter.

File: pontoon/teams/views.py [hunk 3]
```diff
@@ -122,13 +122,11 @@ def ajax_permissions(request, locale):
     translators = l.translators_group.user_set.exclude(pk__in=managers).all()
     all_users = User.objects.exclude(pk__in=managers).exclude(pk__in=translators).exclude(email='')
 
-    contributors = (
-        User.translators
-        .filter(translation__locale=l)
-        .values_list('email', flat=True)
-        .distinct()
-    )
+    contributors_qs = User.translators.with_translation_counts(None, Q(locale=l))
```
Comment (jotes) [inline pontoon/teams/views.py]: [nit/question] From what I understood `with_translation_count` will return only 100 contributors sorted by translation count?
Is that enough for every locale?

Comment (mathjazz) [reply pontoon/teams/views.py]: Good call! I'll fix that!

**Comment by jotes** [comment]
LGTM (I wasn't sure If I have to approve again).

**Pull request merged**